save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestSuitesService.kt

```diff
@@ -16,12 +17,28 @@ class TestSuitesService {
     private lateinit var testSuiteRepository: TestSuiteRepository
 
     /**
-     * @param testSuitesDto
-     * @return list of TestSuites
+     * Save new test suites to DB
+     *
+     * @param testSuitesDto test suites, that should be checked and possibly saved
+     * @return list of *all* TestSuites
      */
+    @Suppress("UnsafeCallOnNullableType")
     fun saveTestSuite(testSuitesDto: List<TestSuiteDto>): List<TestSuite> {
-        val testSuites = testSuitesDto.map { TestSuite(it.type, it.name, it.project, LocalDateTime.now(), "save.properties") }
-        testSuiteRepository.saveAll(testSuites)
-        return testSuites
+        val testSuites = testSuitesDto
+            .map { TestSuite(it.type, it.name, it.project, null, "save.properties") }
+            .map { testSuite ->
+                // try to find TestSuite in the DB based on all non-null properties of `testSuite`
```

Comment:
I think this logic is best done in the database itself. Add a primary key to a bunch of fields: type + name + project + properties

Reply:
You mean like a composite key of four fields? What about id then? And the main advantage of your approach would be that we can call `saveAll` without additional checks before? I think for a relatively small number of queries (there usually will be not many test suites), such change would introduce to much complexity: managing composite key in hibernate, updating DB schema every time we add a new field in this class etc.

Reply:
Yes, I agree. This is probably better